docs/tutorial/quickstart.md

```diff
@@ -0,0 +1,282 @@
+
+```
+ctx := context.Background()
+registry := "myregistry.example.io"
+repository := "myrepo"
+repo, err := remote.NewRepository(fmt.Sprintf("%s/%s", registry, repository))
+if err != nil {
+	panic(err)
+}
+// Note: The below code can be omitted if authentication is not required.
+repo.Client = &auth.Client{
+	Client: retry.DefaultClient,
+	Cache:  auth.NewCache(),
+	Credential: auth.StaticCredential(registry, auth.Credential{
+		Username: "username",
+		Password: "password",
+	}),
+}
+```
+
+## Show tags in the repository
+The following code snippet uses the [(*Repository) Tags](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.Tags) method from the [remote](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote) package to list the tags in the repository.
+
+```
+err = repo.Tags(ctx, "", func(tags []string) error {
+	for _, tag := range tags {
+		fmt.Println(tag)
+	}
+	return nil
+})
+if err != nil {
+	panic(err)
+}
+```
+
+## Push layers to the repository
+
+Before pushing a manifest, the layers it references must already exist in the repository. The following code snippet demonstrates how to push a config layer and a manifest layer using the [(*Repository) Push](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.Push) method.
+
+```
+// push config layer
+configLayer := []byte("example config layer")
+configLayerDescriptor := content.NewDescriptorFromBytes(ocispec.MediaTypeImageLayer, configLayer)
+err = repo.Push(ctx, configLayerDescriptor, bytes.NewReader(configLayer))
+if err != nil {
+	panic(err)
+}
+fmt.Println("Pushed config layer")
+
+// push manifest layer
+manifestLayer := []byte("example manifest layer")
+manifestLayerDescriptor := content.NewDescriptorFromBytes(ocispec.MediaTypeImageLayer, manifestLayer)
+err = repo.Push(ctx, manifestLayerDescriptor, bytes.NewReader(manifestLayer))
+if err != nil {
+	panic(err)
+}
+fmt.Println("Pushed manifest layer")
+```
+
+## Push a manifest to the repository with the tag "quickstart"
+
+The following code snippet demonstrates how to assemble a manifest and push it to the repository with the tag "quickstart" using the [(*Repository) PushReference](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.PushReference) method.
+
+```
+tag := "quickstart"
+manifest := ocispec.Manifest{
+	Versioned: specs.Versioned{
+		SchemaVersion: 2, // historical value. does not pertain to OCI or docker version
+	},
+	MediaType: ocispec.MediaTypeImageManifest,
+	Config:    content.NewDescriptorFromBytes(ocispec.MediaTypeImageConfig, configLayer),
+	Layers:    []ocispec.Descriptor{manifestLayerDescriptor},
```

Comment:
Wondering if it would be clearer to combine the "push layer" and "fetch layer" sections into a "push artifact" section

Reply:
Do you mean "push layer" and "push manifest"?

Reply:
I think it's better to separate "push layer" and "push manifest", as we need to give extra explanation about each operation.